src/rsa/padding.rs

```diff
@@ -63,47 +68,48 @@ impl Encoding for PKCS1 {
 }
 
 impl Verification for PKCS1 {
-    fn verify(&self, msg: untrusted::Input, encoded: untrusted::Input)
-              -> Result<(), error::Unspecified> {
-        encoded.read_all(error::Unspecified, |decoded| {
-            if try!(decoded.read_byte()) != 0 ||
-               try!(decoded.read_byte()) != 1 {
-                return Err(error::Unspecified);
-            }
+    fn verify(&self, msg: untrusted::Input, m: &mut untrusted::Reader,
+              _mod_bits: usize) -> Result<(), error::Unspecified> {
+        let em = m;
 
-            let mut ps_len = 0;
-            loop {
-                match try!(decoded.read_byte()) {
-                    0xff => {
-                        ps_len += 1;
-                    },
-                    0x00 => {
-                        break;
-                    },
-                    _ => {
-                        return Err(error::Unspecified);
-                    },
-                }
-            }
-            if ps_len < 8 {
-                return Err(error::Unspecified);
-            }
+        if try!(em.read_byte()) != 0 ||
+            try!(em.read_byte()) != 1 {
```

Comment:
I've now split it into two conditions so that the code coverage can see whether we're testing both checks.